clash-vexriscv/src/VexRiscv/FFI.hsc

```diff
@@ -174,27 +197,28 @@ instance Storable OUTPUT where
       <*> (#peek OUTPUT, jtag_TDO) ptr
 
     {-# INLINE poke #-}
-    poke ptr this = do
-      (#poke OUTPUT, iBusWishbone_CYC) ptr (iBusWishbone_CYC this)
-      (#poke OUTPUT, iBusWishbone_STB) ptr (iBusWishbone_STB this)
-      (#poke OUTPUT, iBusWishbone_WE) ptr (iBusWishbone_WE this)
-      (#poke OUTPUT, iBusWishbone_ADR) ptr (iBusWishbone_ADR this)
-      (#poke OUTPUT, iBusWishbone_DAT_MOSI) ptr (iBusWishbone_DAT_MOSI this)
-      (#poke OUTPUT, iBusWishbone_SEL) ptr (iBusWishbone_SEL this)
-      (#poke OUTPUT, iBusWishbone_CTI) ptr (iBusWishbone_CTI this)
-      (#poke OUTPUT, iBusWishbone_BTE) ptr (iBusWishbone_BTE this)
-
-      (#poke OUTPUT, dBusWishbone_CYC) ptr (dBusWishbone_CYC this)
-      (#poke OUTPUT, dBusWishbone_STB) ptr (dBusWishbone_STB this)
-      (#poke OUTPUT, dBusWishbone_WE) ptr (dBusWishbone_WE this)
-      (#poke OUTPUT, dBusWishbone_ADR) ptr (dBusWishbone_ADR this)
-      (#poke OUTPUT, dBusWishbone_DAT_MOSI) ptr (dBusWishbone_DAT_MOSI this)
-      (#poke OUTPUT, dBusWishbone_SEL) ptr (dBusWishbone_SEL this)
-      (#poke OUTPUT, dBusWishbone_CTI) ptr (dBusWishbone_CTI this)
-      (#poke OUTPUT, dBusWishbone_BTE) ptr (dBusWishbone_BTE this)
-
-      (#poke OUTPUT, jtag_debug_resetOut) ptr (jtag_debug_resetOut this)
-      (#poke OUTPUT, jtag_TDO) ptr (jtag_TDO this)
+    poke ptr this0 = do
+      this1 <- undefinedToRandom this0
+      (#poke OUTPUT, iBusWishbone_CYC) ptr (iBusWishbone_CYC this1)
+      (#poke OUTPUT, iBusWishbone_STB) ptr (iBusWishbone_STB this1)
```

Comment:
Same comment as before, but do note that (for example) `iBusWishbone_SEL` is represented as a byte, while its hardware type isn't as wide. I'm not sure what bugs in Verilator we'd trigger if bits are set that are usually not used / unset.

Also note that we don't actually use this `poke` instance though. We never transfer this data type from H to C, only from C to H.